src/server/services/procedures/cloud-variables/cloud-variables.js

```diff
@@ -10,6 +10,9 @@ const logger = require('../utils/logger')('cloud-variables');
 const Storage = require('../../storage');
 const Q = require('q');
 
+const globalListeners = {}; // map<var name, map<client id, [socket, msg name]>>
```

Comment:
Why isn't this just `map<var name, [(socket, msg name)]>`? I don't see any reason why we would need random access using the client ID. This would also simplify the `_sendUpdate` logic.

Reply:
Ah, was this to ensure that the same client wasn't added twice? (ie, overwrite is basically a no-op)

Reply:
Right, this was so same clients go in their own bucket to avoid dupe. This'll also be nice with the listen timeout interval since we can just delete the user id once and they'll all go together.